frontend/src/plugins/errors.ts

```diff
@@ -0,0 +1,144 @@
+import axios from "axios"
+import { Plugin, Context } from "@nuxt/types"
+
+import { ERR_UNKNOWN, ErrorCode, errorCodes } from "~/constants/errors"
+import type { FetchingError, RequestKind } from "~/types/fetch-state"
+import type { SupportedSearchType } from "~/constants/media"
+
+import { useAnalytics } from "~/composables/use-analytics"
+
+const isValidErrorCode = (
+  code: string | undefined | null
+): code is ErrorCode => {
+  if (!code) {
+    return false
+  }
+  return (errorCodes as readonly string[]).includes(code)
+}
+
+function isDetailedResponseData(data: unknown): data is { detail: string } {
+  return !!data && typeof data === "object" && "detail" in data
+}
+
+/**
+ * Normalize any error occurring during a network call.
+ *
+ * @param error - Any error arising during a network call
+ * @param searchType - The type of search selected when the error occurred
+ * @param requestKind - The kind of request the error occurred for
+ * @param details - Any additional details to attach to the error
+ * @returns Normalized error object
+ */
+export function normalizeFetchingError(
+  error: unknown,
+  searchType: SupportedSearchType,
+  requestKind: RequestKind,
+  details?: Record<string, string>
+): FetchingError {
+  const fetchingError: FetchingError = {
+    requestKind,
+    details,
+    searchType,
+    code: ERR_UNKNOWN,
+  }
+
+  if (!axios.isAxiosError(error)) {
+    fetchingError.message = (error as Error).message
+    return fetchingError
+  }
+
+  // Otherwise, it's an AxiosError
+  if (isValidErrorCode(error.code)) {
+    fetchingError.code = error.code
+  }
+
+  if (error.response?.status) {
+    fetchingError.statusCode = error.response.status
+  }
+
+  const responseData = error?.response?.data
+
+  // Use the message returned by the API.
+  if (isDetailedResponseData(responseData)) {
+    fetchingError.message = responseData.detail as string
+  } else {
+    fetchingError.message = error.message
+  }
+
+  return fetchingError
+}
+
+/**
+ * Record network errors using the appropriate tool, as needed,
+ * based on response code, status, and request kind.
+ * @param error - The error to record
+ */
+function recordError(
+  context: Context,
+  originalError: unknown,
+  fetchingError: FetchingError
+) {
+  if (fetchingError.code === "ERR_NETWORK") {
+    /**
+     * Record network errors in Plausible so that we can evaluate potential
+     * regional or device configuration issues, for which Sentry is not
+     * as good a tool. Additionally, the number of these events are trivial
+     * for Plausible, but do actually affect our Sentry quota enough that it
+     * is worth diverting them.
+     */
+    const { sendCustomEvent } = useAnalytics()
+    sendCustomEvent("NETWORK_ERROR", {
+      requestKind: fetchingError.requestKind,
+      searchType: fetchingError.searchType,
+    })
```

Comment:
Although we won't lose information by sending them only to Plausible? Will we know which page, for example, the error occurs on?

Reply:
Yes, all of that is automatically recorded for all Plausible events: https://plausible.io/docs/stats-api#properties

All plausible "goal" events (custom events) are essentially just an extension on a regular page view event, with optional additional information on top of the page view props.